agent/run_job.go

```diff
@@ -62,6 +79,37 @@ func (r *JobRunner) Run(ctx context.Context) error {
 		r.cleanup(ctx, wg, exit)
 	}(ctx, &wg) // Note the non-cancellable context (ctx rather than cctx) here - we don't want to be interrupted during cleanup
 
+	ise := &invalidSignatureError{}
+	switch err := r.verifyJob(verifier); {
+	case err == nil: // no error, all good, keep going
+		if verifier != nil {
+			r.logger.Info("Successfully verified job %s with signature %s", r.conf.Job.ID, r.conf.Job.Step.Signature.Value)
+			r.logStreamer.Process([]byte(fmt.Sprintf("✅ Verified job with signature %s\n", r.conf.Job.Step.Signature.Value)))
+		}
+
+	case errors.Is(err, ErrNoSignature):
+		r.verificationFailureLogs(err, r.NoSignatureBehavior)
+		if r.NoSignatureBehavior == VerificationBehaviourBlock {
+			exit.Status = -1
+			exit.SignalReason = "job_verification_failed_no_signature"
+			return nil
+		}
+
+	case errors.As(err, &ise):
+		r.verificationFailureLogs(err, r.InvalidSignatureBehavior)
+		if r.InvalidSignatureBehavior == VerificationBehaviourBlock {
+			exit.Status = -1
+			exit.SignalReason = "job_verification_failed_invalid_signature"
+			return nil
+		}
+
+	default: // some other error
+		r.verificationFailureLogs(err, VerificationBehaviourBlock) // errors in verification are always fatal
+		exit.Status = -1
+		exit.SignalReason = "job_verification_failed_with_error"
+		return nil
+	}
```

Comment:
I do think the `if/else` chain I suggested last time looks better here.

I, too, generally prefer `switches` to `if/else` chains, but here:

1. There is no added efficiency in using a `switch` as all the case statements have to be evaluated in order in the worst case as there is no value to switch on.
2. This statement leaks the `ise` variable. The `if/else` chain would scope it exclusively to the appropriate branch


But I'm not going to die on this hill.

Reply:
If I were to change anything, it would be to replace `default:` with `case err != nil:`, and not make `err == nil` a case in the switch.

As in, the other side of "indent error flow" is "don't indent non-error flow".